src/WebJobs.Script/Rpc/LanguageWorkerChannelInitializationService.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.Azure.WebJobs.Script.Rpc
+{
+    public class LanguageWorkerChannelInitializationService : IHostedService
+    {
+        private readonly IEnumerable<string> _languages = new List<string>()
+        {
+            LanguageWorkerConstants.JavaLanguageWorkerName
+        };
+
+        private readonly string _workerRuntime;
+
+        private ILanguageWorkerChannelManager _languageWorkerChannelManager;
+
+        public LanguageWorkerChannelInitializationService(ILanguageWorkerChannelManager languageWorkerChannelManager)
+        {
+            _languageWorkerChannelManager = languageWorkerChannelManager ?? throw new ArgumentNullException(nameof(languageWorkerChannelManager));
+            _workerRuntime = SystemEnvironment.Instance.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeSettingName);
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            if (SystemEnvironment.Instance.IsLinuxAppServiceEnvironment())
+            {
+                return Task.CompletedTask;
```

Comment:
Don't we need to start the workers for Linux/LinuxContainer as well when in non-placeholder mode? These short circuits seem to mean we won't start anything for Linux (i.e. won't get to the single language clause below).

Reply:
This is intentional. Java is not supported on Linux. Other languages workers are not ready to move out of jobhost

Reply:
I thought this service was the sole place where we're creating and starting these workers. E.g. in Linux Consumption Python Worker, how does the worker get started? I wasn't aware that there was another create path for the worker happening still.